proxy-server/proxy-server.go

```diff
@@ -345,8 +466,10 @@ func handleRequestAndRedirect(res http.ResponseWriter, req *http.Request) {
 	log.Printf("OK: Proxying to url: %s %s\n", tmpTargetURL, req.URL.String())
 }
 
-// Check token validity for the requested resource. Returns true if no error has been thrown.
-// If false is returned, the caller should assume that a response has already been written out.
+/*
+	Check authorization header validity for the requested resource. Returns true if no error has been thrown.
+	If false is returned, the caller should assume that a response has already been written out.
+*/
 func checkTokenAuth(res http.ResponseWriter, req *http.Request, authHeader string) bool {
```

Comment:
@forana Do you know if this auth header verification actually gets used by anything?

Reply:
What do you mean? This is how we make sure that the proxy token for `abc.customer.saturn` is actually for that host (and not something like `def.customer.saturn`).

Reply:
This only gets called when there is an authorization header. Which considering we don't have any control over the backends that use proxy auth is never. And the route it uses to verify the token is `/api/deployments/auth`, which only validates deployments

Reply:
OH. Yes. This was to allow customers to make calls via token instead of the whole login-redirect shenaniganery. Was added in #25

Reply:
And the reason for #25 was to allow deployments to be used from automation. So you could do something like `curl -H "Authorization: Server my-sweet-token" https://thing.mine.saturncloud.org/some/api/call"` without having to pretend to be a browser and deal with token expiration/refreshing.

Reply:
Ah, got it. I will add some comments about that

Reply:
Thank you. That project ended abruptly - there were meant to be tests and customer docs as well :/